### ISSUE TITLE
It's not the same between node an web

### ISSUE THREAD

**Issue by supidea** [mathjax/MathJax-demos-node#8]
ex:
node:
`const renderMath = (texEquation, options = {}) => {
  return new Promise((rs, rj) => {
    mathjax
      .handleRetriesFor(() => {
        return html.convert(texEquation, {
          display: false
        })
      })
      .then(node => rs(adaptor.innerHTML(node)))
      .catch(error => rj(error.message))
  })
}

renderMath('\\d{的}')
  .then(node => {
    console.log(node)
  })
  .catch(error => {
    console.log(error)
  })
`

web
`window.MathJax.tex2svgPromise('\\d{的}', {
          display: false
       } )`

macros
`{d: ['{\\hspace{3pt}{\\scriptsize\\raise-11pt\\bullet} \\hspace{-5pt}{ #1}}', 1]}`

render result

![image](https://user-images.githubusercontent.com/2585305/70891553-66915980-2022-11ea-9d4c-81a2b3041b0d.png)

can you help me

**Comment by dpvc** [comment]
You are correct, the two are not the same.  The reason is that running node on the server means that you don't have an actual DOM to work with, and so can't measure anything, as you can in the browser.  For characters that are in MathJax's font, that is not a problem, because MathJax has the font metrics built in; but for characters that are not in the MathJax fonts, like the `的` that you are using, MathJax doesn't know how wide they are, and can't use the DOM to measure them (as it does in the browser).  So server-side MathJax tries to use a fixed-width font for these, and makes its best guess at the width.  In your case, the character likely is not in the fixed-width font, and MathJax guesses the width incorrectly.

MathJax v3 has better support for adding characters that are not originally in its fonts, but the tools for making the data for that are not yet available; they will be in a future version.  If you were using CommonHTML output, you could manage a few characters by hand, but for SVG output, you would need to produce the outline paths for the characters, and that is not trivial.

An alternative approach would be to use something like Puppeteer in node, which uses a headless Chrome browser (meaning one with no window), which would make an actual DOM available where measurements could be made.  But that would take some work to link into MathJax v3.  It might be easier to run MathJax inside Puppeteer directly to do the conversions just like you would in a regular web page.  You could control that from your main node script.

**Comment by dpvc** [comment]
I'm going to move your issue to mathjax/MathJax, so that it is easier to find via searching the MathJax issues.
